### ISSUE TITLE
[BUG] ApacheTikaDocumentParser doesn't work in production

### ISSUE THREAD

**Issue by florian-h05** [langchain4j/langchain4j#2209]
### The Problem

When using the `ApacheTikaDocumentParser` in production, pulling it in as dependency causes the following code (which is internally used by the Tika parser)

```java
XPathFactory factory = XPathFactory.newInstance();
```

to always fail with:

```
javax.xml.xpath.XPathFactoryConfigurationException:
  No XPathFctory implementation found for the object model:
    http://java.sun.com/jaxp/xpath/dom
```

### How to Reproduce

The issue can be reproduced by adding the dependency on the Tika document parser module:

```xml
        <dependency>
            <groupId>dev.langchain4j</groupId>
            <artifactId>langchain4j-document-parser-apache-tika</artifactId>
            <version>0.36.2</version>
        </dependency>
```

Add the ApacheTikaDocumentParser somewhere in the application code and use it.
Alternative approach: Add the ApacheTikaDocumentParser and add the above code to the application, make sure the above code is executed.

Interestingly, the issue does not occur in Quarkus Dev mode.

### Additional Information

- LangChain4j version: 0.36.2
- Java version: OpenJDK Runtime Environment Temurin-21.0.5+11 (build 21.0.5+11-LTS)
- Quarkus: 3.17.0

**Comment by langchain4j** [comment]
cc @geoand

**Comment by geoand** [comment]
Do you have a sample project we can look at?

**Comment by florian-h05** [comment]
The project where this happened is private, but I can check if I can spin up a minimal project where the issue occurs.

Not sure you if you had a look at the linked PR https://github.com/langchain4j/langchain4j/pull/2208, but the issue was gone with upgrading Apache Tika.

**Comment by florian-h05** [comment]
@geoand I have a minimal reproduce project for this issue: https://github.com/florian-h05/langchain4j-2209.

**Comment by geoand** [comment]
Ok thanks. 

I actually have zero clue of what may be going on as I know nothing about XML or Tika. @sberyozkin does this ring a bell?

**Comment by florian-h05** [comment]
Same ... however did the Tika update to 3.0.0 fix the problem. This update removed the xerces2 dependency, which is related to the XML stuff. But that are just assumptions what could be going on.

**Comment by florian-h05** [comment]
@langchain4j: @geoand has no clue either, how should we proceed? Take my fix from #2208?

**Comment by andreadimaio** [comment]
Good to know that the new version fixes the problem, in the meantime this allowed me to use the production jar.
```xml
<dependency>
	<groupId>dev.langchain4j</groupId>
	<artifactId>langchain4j-document-parser-apache-tika</artifactId>
	<version>0.36.2</version>
	<exclusions>
		<exclusion>
			<groupId>xml-apis</groupId>
			<artifactId>xml-apis</artifactId>
		</exclusion>
	</exclusions>
</dependency>
```
